File: android/src/main/java/io/boxo/expo/CustomEventData.kt
package io.boxo.expo

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class CustomEventData : Record {
    @Field
    val appId: String = ""

    @Field
    val requestId: Int? = null

    @Field
    val type: String? = null

    @Field
    val errorType: String? = null

    @Field
    val payload: Map<String, Any>? = null
}

File: ios/CustomEventData.swift
//
//  CustomEventData.swift
//  ExpoBoxoSdk
//
//  Created by Azamat Kushmanov on 18/9/24.
//

import ExpoModulesCore

struct CustomEventData : Record {
    @Field
    var appId: String = ""

    @Field
    var requestId: Int?

    @Field
    var type: String?

    @Field
    var errorType: String?

    @Field
    var payload: [String : Any]?
}
